invokeai/app/invocations/controlnet_image_processors.py

```diff
@@ -476,37 +479,86 @@ def run_processor(self, image: Image.Image) -> Image.Image:
+        default="regular", description="The Tile ControlNet pre-processing mode to use."
+    )
 
-    # tile_resample copied from sd-webui-controlnet/scripts/processor.py
-    def tile_resample(
-        self,
-        np_img: np.ndarray,
-        res=512,  # never used?
-        down_sampling_rate=1.0,
-    ):
+    # referenced from
+    # https://huggingface.co/TTPlanet/TTPLanet_SDXL_Controlnet_Tile_Realistic/blob/37f1c4575b543fb2036e39f5763d082fdd135318/TTP_tile_preprocessor_v5.py
+    def _apply_gaussian_blur(self, image_np: np.ndarray[Any, Any], ksize: int = 5, sigma_x: float = 1.0):
+        if ksize % 2 == 0:
+            ksize += 1  # ksize must be odd
+        blurred_image = cv2.GaussianBlur(image_np, (ksize, ksize), sigmaX=sigma_x)
+        return blurred_image
+
+    # referenced from
+    # https://huggingface.co/TTPlanet/TTPLanet_SDXL_Controlnet_Tile_Realistic/blob/37f1c4575b543fb2036e39f5763d082fdd135318/TTP_tile_preprocessor_v5.py
+    def _apply_guided_filter(self, image_np: np.ndarray[Any, Any], radius: int, eps: float, scale: int):
+        filter = FastGuidedFilter(image_np, radius, eps, scale)
+        return filter.filter(image_np)
+
+    def _regular_resample(self, np_img: np.ndarray[Any, Any]):
+        height, width, _ = np_img.shape
         np_img = HWC3(np_img)
-        if down_sampling_rate < 1.1:
+        if self.down_sampling_rate < 1.1:
             return np_img
-        H, W, C = np_img.shape
-        H = int(float(H) / float(down_sampling_rate))
-        W = int(float(W) / float(down_sampling_rate))
-        np_img = cv2.resize(np_img, (W, H), interpolation=cv2.INTER_AREA)
+
+        new_height = int(float(height) / float(self.down_sampling_rate))
+        new_width = int(float(width) / float(self.down_sampling_rate))
+        np_img = cv2.resize(np_img, (new_width, new_height), interpolation=cv2.INTER_AREA)
+        return np_img
+
+    # referenced from
+    # https://huggingface.co/TTPlanet/TTPLanet_SDXL_Controlnet_Tile_Realistic/blob/37f1c4575b543fb2036e39f5763d082fdd135318/TTP_tile_preprocessor_v5.py
+    def _blur_resample(self, np_img: np.ndarray[Any, Any]):
+        height, width, _ = np_img.shape
+        ratio = np.sqrt(1024.0 * 1024.0 / (width * height))
+        resize_w, resize_h = int(width * ratio), int(height * ratio)
+        np_img = cv2.resize(np_img, (resize_w, resize_h))
+
+        blur_strength = random.sample([i / 10.0 for i in range(10, 201, 2)], k=1)[0]
+        radius = random.sample([i for i in range(1, 40, 2)], k=1)[0]  # noqa: C416
+        eps = random.sample([i / 1000.0 for i in range(1, 101, 2)], k=1)[0]
+        scale_factor = random.sample([i / 10.0 for i in range(10, 181, 5)], k=1)[0]
+
+        if random.random() > 0.5:
+            np_img = self._apply_gaussian_blur(np_img, ksize=int(blur_strength), sigma_x=blur_strength / 2)
+
+        if random.random() > 0.5:
+            np_img = self._apply_guided_filter(np_img, radius, eps, int(scale_factor))
```

Comment:
All of the random sampling in this block makes it look like experimental code?
Nodes should be deterministic, so we need to either:
- Hard-code these values to good defaults.
- Expose them as parameters.
- Add a seed. (This seems like a bad idea. I can't think of a good reason for a preprocessor to need a seed.)